olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -36,7 +36,56 @@ namespace read {
 class VersionedLayerClientImpl;
 
 /**
- * @brief The VersionedLayerClient aimed to acquire data from OLP services.
+ * @brief The VersionedLayerClient is aimed at acquiring data from a versioned
+ * layer of the OLP. The versioned layer stores slowly-changing data that must
+ * remain logically consistent with other layers in the catalog.
+ *
+ * @note When you request a particular version of data from a versioned layer,
+ * the partition that gets returned may have a lower version number than you
+ * requested. Only those layers and partitions that are updated have their
+ * version updated to the catalog's new version number, the version of a
+ * layer or partition represents the catalog version in which the layer or
+ * partition was last updated.
+ *
+ * @note When version is not specified, the latest is used. To query the latest
+ * version of catalog \b additional request is performed.
+ *
+ * Example with version provided, which saves one network request:
+ * @code{.cpp}
+ * const std::string kCatalog = "hrn:here:data:::here-optimized-map-for-visualization-2;
+ * const std::string kLayerId = "omv-base-v2";
+ * const auto kHRN = olp::client::HRN::FromString(kCatalog);
+ * const auto kPartitionId = "269";
+ * const auto kVersion = 100;
+ *
+ * std::shared_ptr<olp::thread::TaskScheduler> task_scheduler =
+ *    olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(1u);
+ * std::shared_ptr<olp::http::Network> http_client = olp::client::
+ *    OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler();
+ *
+ * olp::authentication::Settings settings;
+ * settings.task_scheduler = task_scheduler;
+ * settings.network_request_handler = http_client;
+ *
+ * olp::client::AuthenticationSettings auth_settings;
+ * auth_settings.provider = olp::authentication::TokenProviderDefault(
+ *     kKeyId, kKeySecret, std::move(settings));
+ *
+ * auto client_settings = olp::client::OlpClientSettings();
+ * client_settings.authentication_settings = auth_settings;
+ * client_settings.task_scheduler = std::move(task_scheduler);
+ * client_settings.network_request_handler = std::move(http_client);
+ *
+ * VersionedLayerClient client{kHRN, kLayerId, client_settings};
+ * VersionedLayerClient::Callback<model::Data> callback = ... ;
```

Comment:
Maybe instead of "..." provide an empty lambda, so that readers might see that the response is a ApiResponse<Data, Error>.